### DISCUSSION TITLE
Imputed module score values

### DISCUSSION THREAD

**Discussion by peggiechien** [GreenleafLab/ArchR#1053] [Questions / Documentation]
Hello, is it possible to retrieve the imputed values of a module score?

**Comment by rcorces** [comment]
No. Imputation is performed on the fly and imputed values / matrices are not stored. You theoretically could use the `imputeMatrix()` function to do this on your own.

**Comment by peggiechien** [comment]
Got it, thank you. After performing `imputeMatrix()`, is it possible to then add the imputed matrix into the original ArchRProject as a new custom matrix?

**Reply by rcorces** [reply]
unfortunately no. This is a feature that has been requested before and its on our radar. Its challenging for various reasons.

**Reply by rcorces** [reply]
Though module scores arent stored as a matrix - they are stored in cellColData unless I'm mistaken. So you might be able to add them in manually. I'd need to go back and look at the code.

**Reply by peggiechien** [reply]
Yes, the module scores are in cellColData. Though after performing `imputeMatrix()`, I'm not sure how to produce imputed module score values without adding the imputed matrix into the original ArchRProject so that the imputed matrix can be used for useMatrix in `addModuleScore()`.

**Reply by rcorces** [reply]
I see. unfortunately I dont think this is currently possible

**Reply by peggiechien** [reply]
Got it, thank you
